src/content_mapblock.cpp

```diff
@@ -401,9 +405,15 @@ void mapblock_mesh_generate_special(MeshMakeData *data,
 
 			// Neighbor liquid levels (key = relative position)
 			// Includes current node
-			std::map<v3s16, f32> neighbor_levels;
-			std::map<v3s16, content_t> neighbor_contents;
-			std::map<v3s16, u8> neighbor_flags;
+
+			struct NeighbourData
+			{
+				f32 level;
+				content_t content;
+				u8 flags;
+			};
+			NeighbourData neighbour_datas[27];
```

Comment:
Except that would clash with another variable called neighbour_data. I would have happily used it otherwise. I'm not sure there are a good set of two names. My mental parsing of neighbour_datas is (neighbour_data)s. Do you have suggestions for both.